src/validation.cpp

```diff
@@ -4973,24 +4973,6 @@ CBlockFileInfo* GetBlockFileInfo(size_t n)
     return &vinfoBlockFile.at(n);
 }
 
-ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::DeploymentPos pos)
-{
-    LOCK(cs_main);
```

Comment:
This global was introduced by d23f6c6. At that commit, the global`VersionBitsTipState`'s unique caller`BIP9SoftForkDesc` didn't require `cs_main` locking. 

Those methods were refactored by 3862e47, where the new `BIP9SoftForkDescPushBack` did require the lock. I think since then this lock has been useless and could have been substituted by an annotation.

So not a concern to get rid of them.